File: app/src/main/java/com/example/vijay/roomdatabaseexample/MainActivity.java
package com.example.vijay.roomdatabaseexample;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.vijay.roomdatabaseexample.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;
    DatabaseHandler db = new DatabaseHandler(this);
    EmpDetails empDetails;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);


        binding.addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View src) {

                int id = binding.idEdtEmpId.getId();
                String name = binding.idEdtEmpName.getText().toString();
                String role = binding.idEdtEmpRole.getText().toString();
                String org = binding.idEdtEmpOrg.getText().toString();

                empDetails = new EmpDetails(id, name, role, org);
                if (empDetails != null) {
                    db.addEmployee(empDetails);
                } else {
                    Toast.makeText(MainActivity.this, "please check enter proper employee details ", Toast.LENGTH_SHORT).show();
                }


            }
        });
    }

}

File: app/src/main/java/com/example/vijay/firebasecloudmessaging/MainActivity.kt
package com.example.vijay.firebasecloudmessaging

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import java.io.IOException

class MainActivity : AppCompatActivity() {
    private val senderID = "YOUR_SENDER_ID"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        getToken()
    }
//    private fun getToken() {
//
//        Thread(Runnable {
//            try {
//
//                val newToken = FirebaseInstanceId.getInstance()
//                    .getToken(senderID, "FCM")
//                println("Token --> $newToken")
//
//            } catch (e: IOException) {
//                e.printStackTrace()
//            }
//        }).start()
//    }

}